packages/expo-google-sign-in/android/src/main/java/expo/modules/google/signin/GoogleSignInModule.java

```diff
@@ -0,0 +1,449 @@
+        scopes.put("FITNESS_BLOOD_PRESSURE_READ_WRITE", Scopes.FITNESS_BLOOD_PRESSURE_READ_WRITE);
+        scopes.put("FITNESS_BLOOD_GLUCOSE_READ", Scopes.FITNESS_BLOOD_GLUCOSE_READ);
+        scopes.put("FITNESS_BLOOD_GLUCOSE_READ_WRITE", Scopes.FITNESS_BLOOD_GLUCOSE_READ_WRITE);
+        scopes.put("FITNESS_OXYGEN_SATURATION_READ", Scopes.FITNESS_OXYGEN_SATURATION_READ);
+        scopes.put("FITNESS_OXYGEN_SATURATION_READ_WRITE", Scopes.FITNESS_OXYGEN_SATURATION_READ_WRITE);
+        scopes.put("FITNESS_BODY_TEMPERATURE_READ", Scopes.FITNESS_BODY_TEMPERATURE_READ);
+        scopes.put("FITNESS_BODY_TEMPERATURE_READ_WRITE", Scopes.FITNESS_BODY_TEMPERATURE_READ_WRITE);
+        scopes.put("FITNESS_REPRODUCTIVE_HEALTH_READ", Scopes.FITNESS_REPRODUCTIVE_HEALTH_READ);
+        scopes.put("FITNESS_REPRODUCTIVE_HEALTH_READ_WRITE", Scopes.FITNESS_REPRODUCTIVE_HEALTH_READ_WRITE);
+        scopes.put("DISPLAY_ADS", Scopes.DISPLAY_ADS);
+        scopes.put("YOUTUBE_DATA_API", Scopes.YOUTUBE_DATA_API);
+
+        final Map<String, Object> signInTypes = new HashMap<>();
+        scopes.put("DEFAULT", "default");
+        scopes.put("GAMES", "games");
+
+
+        constants.put("SCOPES", scopes);
+        constants.put("ERRORS", errors);
+        constants.put("TYPES", signInTypes);
+
+        return constants;
+    }
+
+    @ExpoMethod
+    public void hasPlayServicesAsync(boolean shouldUpdate, Promise promise) {
+        Activity activity = getCurrentActivity();
+
+        if (activity == null) {
+            promise.reject(ERROR_EXCEPTION, "Activity is null");
+            return;
+        }
+
+        GoogleApiAvailability googleApiAvailability = GoogleApiAvailability.getInstance();
+        int status = googleApiAvailability.isGooglePlayServicesAvailable(activity);
+
+        if (status != ConnectionResult.SUCCESS) {
+            if (shouldUpdate && googleApiAvailability.isUserResolvableError(status)) {
+                googleApiAvailability.getErrorDialog(activity, status, RC_PLAY_SERVICES).show();
+            }
+            promise.reject(ERROR_EXCEPTION, "Play Services are not available");
+        } else {
+            promise.resolve(true);
+        }
+    }
+
+    @ExpoMethod
+    public void getCurrentUserAsync(Promise promise) {
+        GoogleSignInAccount account = GoogleSignIn.getLastSignedInAccount(getApplicationContext());
+        if (account == null) {
+            promise.resolve(null);
+            return;
+        }
+        promise.resolve(jsonFromGoogleUser(account));
+    }
+
+    @ExpoMethod
+    public void getPhotoAsync(Number size, Promise promise) {
+        GoogleSignInAccount account = GoogleSignIn.getLastSignedInAccount(getApplicationContext());
+        if (account == null || account.getPhotoUrl() == null) {
+            // TODO: Bacon: Should we error out?
+//            promise.reject(ERROR_EXCEPTION, "Could not get photoURL");
+            promise.resolve(null);
```

Comment:
fwiw I think this is the correct behavior -- it's probably expected that some users will not have a photo specified and I don't think we should reject a promise except in the case of a developer error or weird/inconsistent state